h2/src/main/org/h2/fulltext/FullText.java

```diff
@@ -1095,13 +1155,17 @@ private String getKey(Object[] row) throws SQLException {
             return buff.toString();
         }
 
+        private PreparedStatement getStatement(Connection conn, int indx) throws SQLException {
+            return useOwnConnection ? conn.prepareStatement(SQL[indx]) : prepStatements[indx];
+        }
+
     }
 
     /**
      * INTERNAL
      * Close all fulltext settings, freeing up memory.
      */
-    public static void closeAll() {
+    public static synchronized void closeAll() {
```

Comment:
FullTextSettings.closeAll appears to handle it's own synch needs, no need to mark this method as synch